check_manifest.py

```diff
@@ -533,21 +535,29 @@ def add_directories(names):
                                     r'recursive-include \1 *.\2'),
 ]]
 
+CFG_SECTION_CHECK_MANIFEST = 'check-manifest'
+CFG_IGNORE_DEFAULT_RULES = (CFG_SECTION_CHECK_MANIFEST, 'ignore-default-rules')
+CFG_IGNORE = (CFG_SECTION_CHECK_MANIFEST, 'ignore')
+CFG_IGNORE_BAD_IDEAS = (CFG_SECTION_CHECK_MANIFEST, 'ignore-bad-ideas')
+
 
 def read_config():
     """Read configuration from setup.cfg."""
     # XXX modifies global state, which is kind of evil
+    ignore_bad_ideas = []
     config = ConfigParser.ConfigParser()
     config.read(['setup.cfg'])
-    if not config.has_section('check-manifest'):
+    if not config.has_section(CFG_SECTION_CHECK_MANIFEST):
         return
-    if (config.has_option('check-manifest', 'ignore-default-rules')
-            and config.getboolean('check-manifest', 'ignore-default-rules')):
+    if (config.has_option(*CFG_IGNORE_DEFAULT_RULES)
+            and config.getboolean(*CFG_IGNORE_DEFAULT_RULES)):
         del IGNORE[:]
-    if config.has_option('check-manifest', 'ignore'):
-        patterns = [p.strip() for p in config.get('check-manifest',
-                                                  'ignore').splitlines()]
+    if config.has_option(*CFG_IGNORE):
+        patterns = [p.strip() for p in config.get(*CFG_IGNORE).splitlines()]
         IGNORE.extend(p for p in patterns if p)
+    if config.has_option(*CFG_IGNORE_BAD_IDEAS):
+        lines = config.get(*CFG_IGNORE_BAD_IDEAS).splitlines()
+        IGNORE_BAD_IDEAS.extend(line.strip() for line in lines if line)
```

Comment:
The check ought to be `if line.strip()` to skip lines consisting entirely of whitespace, for consistency with CFG_IGNORE.

(This is why I first strip them into a list before filtering blank ones, in the CFG_IGNORE handling: to avoid having to call .strip() twice on the same string.)